src/Adapter/MySQL.php

```diff
@@ -406,7 +414,12 @@ private function computeFieldName($fieldName, $filterToTableMapping)
             )
         ) {
             $joinMapping = $filterToTableMapping[$fieldName];
-            $fieldName = $joinMapping['tableAlias'] . '.' . (isset($joinMapping['fieldName']) ? $joinMapping['fieldName'] : $fieldName);
+
+            if ($sortByField) {
+                $fieldName = $joinMapping['tableAlias'] . '.' . (isset($joinMapping['fieldName']) ? $joinMapping['fieldName'] : $fieldName);
+            } else {
+                $fieldName = $joinMapping['tableAlias'] . '.' . (isset($joinMapping['fieldName']) ? $joinMapping['fieldName'] : $fieldName) . (isset($joinMapping['fieldAlias']) ? ' as ' . $joinMapping['fieldAlias'] : '');
+            }
```

Comment:
To prevent duplicates.
```suggestion
            $fieldName = $joinMapping['tableAlias'] . '.' . (isset($joinMapping['fieldName']) ? $joinMapping['fieldName'] : $fieldName);
            if ($sortByField === false) {
                $fieldName .= isset($joinMapping['fieldAlias']) ? ' as ' . $joinMapping['fieldAlias'] : '';
            }
```